src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -122,6 +125,66 @@ protected function renderClose($component) {
 		return $tpl->get();
 	}
 
+	protected function renderToggle(Component\Button\Toggle $component) {
+		$tpl = $this->getTemplate("tpl.toggle.html", true, true);
+
+		$on_action = $component->getActionOn();
+		$off_action = $component->getActionOff();
+
+		$on_url = (is_string($on_action))
+			? $on_action
+			: "";
+
+		$off_url = (is_string($off_action))
+			? $off_action
+			: "";
+
+		$signals = [];
+
+		foreach ($component->getTriggeredSignals() as $s)
+		{
+			$signals[] = [
+				"signal_id" => $s->getSignal()->getId(),
+				"event" => $s->getEvent(),
+				"options" => $s->getSignal()->getOptions()
+			];
+
+		}
+
+		$signals = json_encode($signals);
+
+		if ($component->isActive()) {
+			$component = $component->withAdditionalOnLoadCode(function ($id)
+				use ($on_url, $off_url, $signals) {
+				$code = "$('#$id').on('click', function(event) {
```

Comment:
Why did you not use getTriggeredSignalsFor('click') above?

Reply:
Because our customr "toggle_on" and "toggle_off" events are triggered by the HTML "click" event. There is no "toggle_on" event in HTML.